src/fsharp/import.fs

```diff
@@ -287,14 +287,15 @@ let rec ImportProvidedType (env: ImportMap) (m: range) (* (tinst: TypeInst) *) (
                 tcref, [] 
         
         /// Adjust for the known primitive numeric types that accept units of measure. 
-        let tcref = 
-            if tyconRefEq g tcref g.system_Double_tcref && genericArgs.Length = 1 then g.pfloat_tcr
-            elif tyconRefEq g tcref g.system_Single_tcref && genericArgs.Length = 1 then g.pfloat32_tcr
-            elif tyconRefEq g tcref g.system_Decimal_tcref && genericArgs.Length = 1 then g.pdecimal_tcr
-            elif tyconRefEq g tcref g.system_Int16_tcref && genericArgs.Length = 1 then g.pint16_tcr
-            elif tyconRefEq g tcref g.system_Int32_tcref && genericArgs.Length = 1 then g.pint_tcr
-            elif tyconRefEq g tcref g.system_Int64_tcref && genericArgs.Length = 1 then g.pint64_tcr
-            elif tyconRefEq g tcref g.system_SByte_tcref && genericArgs.Length = 1 then g.pint8_tcr
+        let tcref =
+            if genericArgs.Length <> 1 then tcref
```

Comment:
Couldn't we do a pattern match on `genericArgs` and test for a single element?
```fsharp
match genericArgs with
| [_] ->
    if tyconRefEq g ... then ...
    else tcref
| _ -> tcref
```

Reply:
yes. tbh I really wished for List.isSingleton here. But I'm letting this one for another day since genericArgs is usally very very small. the real optimization here is that if Length <> 1 then we don't have to do all those tyconRefEq

Reply:
ok fuck it. I changed it to pattern match, because why not

Reply:
I do understand where you are at. The pattern match is better in the sense that you don't have to iterate over everything. Though, the code below eventually calls `genericArgs.Length` anyway.. Maybe in a perfect world, `genericArgs` shouldn't be a list?